src/pages/[platform]/build-a-backend/auth/multi-step-sign-in/index.mdx

```diff
@@ -20,6 +20,512 @@ export function getStaticProps(context) {
+	const result = await confirmSignIn({ challengeResponse: totpCode });
+
+	return handleSignInResult(result);
+}
+
+```
+
+## Continue signin with MFA Setup Selection
+
+If the next step is `CONTINUE_SIGN_IN_WITH_MFA_SETUP_SELECTION`, then the user must indicate which of the available MFA methods they would like to setup. After the user selects an MFA method to setup, your implementation must pass the selected MFA method to the `confirmSignIn` API.
+
+The MFA types which are currently supported by Amplify Auth for setup are:
+
+- `TOTP`
+- `EMAIL`
+
+Once Amplify receives the users selection, you can expect to handle a follow up `nextStep` corresponding with the selected MFA type for setup:
+- If `EMAIL` is selected, `CONTINUE_SIGN_IN_WITH_EMAIL_SETUP` will be the next step.
+- If `TOTP` is selected, `CONTINUE_SIGN_IN_WITH_TOTP_SETUP` will be the next step.
+
+```ts
+async function handleSignInResult(result: SignInOutput) {
+	switch (result.nextStep.signInStep) {
+		case 'CONTINUE_SIGN_IN_WITH_MFA_SETUP_SELECTION': {
+			const { allowedMFATypes } = result.nextStep;
+			// Present available MFA options to user
+			// Prompt for selection
+			console.log(`There are multiple MFA options available for setup.`);
+			console.log(`Select an MFA type from the allowedMFATypes list.`);
+			break;
+		}
+	}
+}
+
+type MfaType = 'SMS' | 'TOTP' | 'EMAIL';
+
+async function handleMfaSelection(mfaType: MfaType) {
+	const result = await confirmSignIn({ challengeResponse: mfaType });
+
+	return handleSignInResult(result);
+}
+
+```
+
+## Confirm signin with custom challenge
+
+If the next step is `CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE`, Amplify Auth is awaiting completion of a custom authentication challenge. The challenge is based on the AWS Lambda trigger you configured as part of a custom sign in flow.
+
+For example, your custom challenge Lambda may pass a prompt to the frontend which requires the user to enter a secret code.
+
+```ts
+async function handleSignInResult(result: SignInOutput) {
+	switch (result.nextStep.signInStep) {
+		case 'CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE': {
+			const params = result.nextStep.additionalInfo;
+			const hint = params.hint!;
+			// Prompt user to enter custom challenge response
+			console.log(hint); // `Enter the secret code`
+			break;
+		}
+	}
+}
+
+```
+
+To complete this step, you should prompt the user for the custom challenge answer, and pass the answer to the `confirmSignIn` API.
+
+```ts
+async function confirmCustomChallenge(answer: string) {
+	const result = await confirmSignIn({ challengeResponse: answer });
+
+	return handleSignInResult(result);
+}
+```
+
+<Callout warning>
+
+**Special Handling on `confirmSignIn`**
+
+If `failAuthentication=true` is returned by the Lambda, Cognito will invalidate the session of the request. This is represented by a `NotAuthorizedException` and requires restarting the sign-in flow by calling `signIn` again.
+
+</Callout>
+
+## Confirm signin with new password
```

Comment:
signin -> sign-in

Reply:
there are similar topics in 
https://docs.amplify.aws/react/build-a-backend/auth/connect-your-frontend/sign-in/
https://docs.amplify.aws/react/build-a-backend/auth/manage-users/manage-passwords/

Reply:
Referenced the similar topics [here](https://github.com/aws-amplify/docs/pull/8023/commits/c61ad3e8f89e355217ad32b4349a1c9f02122695)